transport/cloudservice.go

```diff
@@ -0,0 +1,173 @@
+package microservicetransport
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/LUSHDigital/microservice-core-golang/response"
+	"github.com/LUSHDigital/microservice-transport-golang/config"
+	"github.com/LUSHDigital/microservice-transport-golang/domain"
+	transportErrors "github.com/LUSHDigital/microservice-transport-golang/errors"
+	"github.com/LUSHDigital/microservice-transport-golang/models"
+)
+
+// AuthCredentials - Credentials needed to authenticate for a cloud service.
+type AuthCredentials struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+// CloudService - Responsible for communication with a cloud service.
+type CloudService struct {
+	Service                      // Inherit all properties of a normal service.
+	Credentials *AuthCredentials // Authentication credentials for cloud service calls.
+	Client      *http.Client
+}
+
+// NewCloudService - Prepare a new CloudService struct with the provided parameters.
+func NewCloudService(client *http.Client, branch, env, namespace, name string, credentials *AuthCredentials) *CloudService {
+	return &CloudService{
+		Service: Service{
+			Branch:      branch,
+			Environment: env,
+			Namespace:   namespace,
+			Name:        name,
+			Client:      client,
+		},
+		Client:      client,
+		Credentials: credentials,
+	}
+}
+
+// authenticate - Authenticate against the API gateway and return an auth token.
+func (c *CloudService) authenticate(request *Request) (*models.Token, error) {
+	loginBody := new(bytes.Buffer)
+	if err := json.NewEncoder(loginBody).Encode(c.Credentials); err != nil {
```

Comment:
Probably should use `json.Marshal` instead. Totally appreciate you're porting code verbatim here, but worth doing while we have the chance imo :)

Reply:
done